src/Engine/RulesStateInterface.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Engine\RulesStateInterface.
+ */
+
+namespace Drupal\rules\Engine;
+
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Plugin\Context\ContextInterface;
+use Drupal\rules\Exception\RulesEvaluationException;
+
+/**
+ * Defines an interface for the rules state.
+ */
+interface RulesStateInterface {
+
+  /**
+   * Adds the given variable to the given execution state.
+   *
+   * @param string $name
+   *   The varible name.
+   * @param \Drupal\Core\Plugin\Context\ContextInterface $context
+   *   The variable wrapped as context.
+   */
+  public function addVariable($name, ContextInterface $context);
+
+  /**
+   * Gets a variable.
+   *
+   * @param string $name
+   *   The name of the variable to return.
+   *
+   * @return \Drupal\Core\Plugin\Context\ContextInterface
+   *   The variable wrapped as context.
+   *
+   * @throws RulesEvaluationException
+   *   Throws a RulesEvaluationException if the variable does not exist in the
+   *   state.
+   */
+  public function getVariable($name);
+
+  /**
+   * Returns a value as specified in the selector.
+   *
+   * @param string $selector
+   *   The selector string, e.g. "node:uid:entity:mail:value".
+   * @param string $langcode
+   *   (optional) The language code used to get the argument value if the
+   *   argument value should be translated. Defaults to
+   *   LanguageInterface::LANGCODE_NOT_SPECIFIED.
+   *
+   * @return \Drupal\Core\TypedData\TypedDataInterface
+   *   The variable wrapped as typed data.
+   *
+   * @throws RulesEvaluationException
```

Comment:
@klausi just made requested changes, thanks for feedback.
Should I "squash" commits into a single one?